infra/bisector.py

```diff
@@ -113,40 +111,52 @@ def bisect(commit_old, commit_new, testcase, fuzz_target, build_data):
-                                                    local_store_path,
-                                                    commit_old)
-  bisect_repo_manager = repo_manager.RepoManager(repo_url, local_store_path)
-  commit_list = bisect_repo_manager.get_commit_list(commit_old, commit_new)
-  build_specified_commit.build_fuzzer_from_commit(
-      build_data.project_name, commit_list[0], bisect_repo_manager.repo_dir,
-      build_data.engine, build_data.sanitizer, build_data.architecture,
-      bisect_repo_manager)
-  error_code = helper.reproduce_impl(build_data.project_name, fuzz_target,
-                                     False, [], [], testcase)
-  old_idx = len(commit_list) - 1
-  new_idx = 0
-  if len(commit_list) == 1:
-    if not error_code:
-      return None
-    return commit_list[0]
-
-  while old_idx - new_idx != 1:
-    curr_idx = (old_idx + new_idx) // 2
+  with tempfile.TemporaryDirectory() as tmp_dir:
+    repo_url, repo_name = build_specified_commit.detect_main_repo_from_docker(
+        build_data.project_name, commit_old)
+    if not repo_url or not repo_name:
+      raise ValueError('Main git repo can not be determined.')
+
+    bisect_repo_manager = repo_manager.RepoManager(
+        repo_url, tmp_dir, repo_name=repo_name)
+    commit_list = bisect_repo_manager.get_commit_list(commit_old, commit_new)
     build_specified_commit.build_fuzzer_from_commit(
-        build_data.project_name, commit_list[curr_idx],
-        bisect_repo_manager.repo_dir, build_data.engine, build_data.sanitizer,
-        build_data.architecture, bisect_repo_manager)
-    error_exists = (
-        helper.reproduce_impl(build_data.project_name, fuzz_target, False, [],
-                              [], testcase) == error_code)
-    if error_exists == error_code:
-      new_idx = curr_idx
-    else:
-      old_idx = curr_idx
-  return commit_list[new_idx]
+        build_data.project_name, commit_list[0], bisect_repo_manager.repo_dir,
+        build_data.engine, build_data.sanitizer, build_data.architecture,
+        bisect_repo_manager)
+    initial_error_code = helper.reproduce_impl(build_data.project_name,
+                                               fuzz_target, False, [], [],
+                                               testcase)
+
+    old_idx = len(commit_list) - 1
+    new_idx = 0
+    while old_idx - new_idx > 1:
+      curr_idx = (old_idx + new_idx) // 2
+      build_specified_commit.build_fuzzer_from_commit(
+          build_data.project_name, commit_list[curr_idx],
+          bisect_repo_manager.repo_dir, build_data.engine, build_data.sanitizer,
+          build_data.architecture, bisect_repo_manager)
+      error_code = helper.reproduce_impl(build_data.project_name, fuzz_target,
+                                         False, [], [], testcase)
+      if initial_error_code == error_code:
+        new_idx = curr_idx
+      else:
+        old_idx = curr_idx
+    if old_idx - new_idx == 1:
```

Comment:
poor GitHub removed our discussion from https://github.com/google/oss-fuzz/pull/3152#pullrequestreview-334739361

if you're concerned that all commits in the range may expose the same behavior, I'd prefer testing `commit_list[old_idx]` before the `while` loop and erroring out if `error_code == initial_error_code` with a message like "Both old and new commits reproduce the same behavior, bisection is not possible".

Reply:
and then these lines 149-158 should be fine to remove

Reply:
Done.